python/pyarrow/_compute.pyx

```diff
@@ -690,99 +678,94 @@ class CastOptions(_CastOptions):
 
 
 cdef class _ElementWiseAggregateOptions(FunctionOptions):
-    def _set_options(self, bint skip_nulls):
+    def _set_options(self, skip_nulls):
         self.wrapped.reset(new CElementWiseAggregateOptions(skip_nulls))
 
 
 class ElementWiseAggregateOptions(_ElementWiseAggregateOptions):
-    def __init__(self, bint skip_nulls=True):
+    def __init__(self, *, skip_nulls=True):
         self._set_options(skip_nulls)
 
 
 cdef CRoundMode unwrap_round_mode(round_mode) except *:
-    if round_mode == 'down':
+    # Explicit if-else since Cython does not supports storing C++ enum classes
+    # as dictionary values because they do not resolve to integers.
+    if round_mode == "down":
         return CRoundMode_DOWN
-    elif round_mode == 'up':
+    elif round_mode == "up":
         return CRoundMode_UP
-    elif round_mode == 'towards_zero':
+    elif round_mode == "towards_zero":
         return CRoundMode_TOWARDS_ZERO
-    elif round_mode == 'towards_infinity':
+    elif round_mode == "towards_infinity":
         return CRoundMode_TOWARDS_INFINITY
-    elif round_mode == 'half_down':
+    elif round_mode == "half_down":
         return CRoundMode_HALF_DOWN
-    elif round_mode == 'half_up':
+    elif round_mode == "half_up":
         return CRoundMode_HALF_UP
-    elif round_mode == 'half_towards_zero':
+    elif round_mode == "half_towards_zero":
         return CRoundMode_HALF_TOWARDS_ZERO
-    elif round_mode == 'half_towards_infinity':
+    elif round_mode == "half_towards_infinity":
         return CRoundMode_HALF_TOWARDS_INFINITY
-    elif round_mode == 'half_to_even':
+    elif round_mode == "half_to_even":
         return CRoundMode_HALF_TO_EVEN
-    elif round_mode == 'half_to_odd':
+    elif round_mode == "half_to_odd":
         return CRoundMode_HALF_TO_ODD
     else:
-        raise ValueError('"{}" is not a valid round mode'.format(round_mode))
+        raise ValueError(f"\"{round_mode}\" is not a valid 'round mode'")
 
 
 cdef class _RoundOptions(FunctionOptions):
-    def _set_options(self, int64_t ndigits, round_mode):
-        cdef:
-            CRoundMode c_round_mode = CRoundMode_HALF_TO_EVEN
-        c_round_mode = unwrap_round_mode(round_mode)
-        self.wrapped.reset(new CRoundOptions(ndigits, c_round_mode))
+    def _set_options(self, ndigits, round_mode):
+        self.wrapped.reset(new CRoundOptions(ndigits,
+                                             unwrap_round_mode(round_mode)))
 
 
 class RoundOptions(_RoundOptions):
-    def __init__(self, ndigits=0, round_mode='half_to_even'):
+    def __init__(self, *, ndigits=0, round_mode="half_to_even"):
```

Comment:
This change doesn't look desirable to me. The Python build-in [round](https://docs.python.org/3/library/functions.html#round) function takes  positional arguments.